x-pack/plugins/apm/common/agent_explorer.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AgentName } from '../typings/es_schemas/ui/fields/agent';
+
+export interface AgentExplorerListItem {
```

Comment:
`ValuesType<APIReturnType<'GET /internal/apm/agent_explorer'>>` could be used instead of this interface so the data contract is only defined in the server

Reply:
I like this approach ❤️